### PR TITLE
Create a new utility package

### PR DESCRIPTION
**Pull request by onursumer**
Fix cbioportal/cbioportal/issues/7371
Related to knowledgesystems/msk-insight/issues/49

Changes:
- Move certain pure utility functions into a separate package for easier export in external projects. Current packages are not compatible with plain node applications due to some DOM dependencies.
- Make the mutation mapper data fetcher customizable to be able to tweak data source for specific data sources. This is useful, for example, when we want to replace an API call with a static data source.

## Checks
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -133,6 +133,7 @@
     "bundle-loader": "^0.5.4",
     "cbioportal-frontend-commons": "^0.2.7",
     "cbioportal-ts-api-client": "^0.9.0-beta.0",
+    "cbioportal-utils": "^0.1.0-beta.0",
```
Comment (onursumer) [inline package.json:136]: Not sure if this is the best name for the new package. Name suggestions are welcome!

File: packages/cbioportal-utils/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "test": "cross-env CI=1 react-scripts-ts test --env=jsdom",
+    "test:watch": "yarn run test -- --watch"
+  },
+  "peerDependencies": {
+    "mobx": "^3.0.0 || ^4.0.0 || ^5.0.0"
+  },
+  "dependencies": {
+    "genome-nexus-ts-api-client": "^1.1.2-beta.0",
+    "lodash": "^4.17.11",
+    "typescript": "3.2.2"
```
Comment (onursumer) [inline packages/cbioportal-utils/package.json:37]: we should update this to 3.8.3 after merging #3171

File: package.json (hunk 1 above)
Comment (zhx828) [reply package.json:136]: Since we have cbioportal-frontend-commons, maybe cbioportal-frontend-utils?

Comment (onursumer) [reply package.json:136]: I was thinking that too, but this package is intended to be used by node scripts/backend, too. So, "frontend" in the package name might be misleading.

**Pull request merged**